BASE/src/ServerTelemetryChannel/Implementation/TelemetrySerializer.cs

```diff
@@ -27,6 +29,27 @@ public Uri EndpointAddress
         }
 
         public virtual void Serialize(ICollection<ITelemetry> items)
+        {
+            this.HandleTelemetryException(items);
+
+            var transmission = new Transmission(this.EndpointAddress, items);
+            this.Transmitter.Enqueue(transmission);
+        }
+
+        public virtual Task<bool> SerializeAsync(ICollection<ITelemetry> items, CancellationToken cancellationToken)
+        {
+            this.HandleTelemetryException(items);
+
+            var transmission = new Transmission(this.EndpointAddress, items, true);
+
+            using (cancellationToken.Register(() => transmission.SetFlushTaskCompletionSourceResult(false)))
+            {
+                Task.Run(() => this.Transmitter.Enqueue(transmission));
```

Comment:
I'd prefer to see cancellation result in a canceled task (not a completed one with a 'false' return).
That would imply that you need a method like `CancelFlushTask()` that calls `SetCanceled()` on the TCS.
That way, await-int the task will throw a TaskCanceledException as expected.
You might also want to pass the cancellationToken into the Task.Run call.
You should also add a `cancellationToken.ThrowIfCanceled()` call at the top of this method.
I'd also like to see unit tests that cover this case.
(Sorry for all the demands)